src/apis/signUp.jsx

```diff
@@ -0,0 +1,26 @@
+// import axios from "axios";
+// import { signUp } from "../urls/index";
+
+// export const fetchSignUp = (e) => {
+//   axios
+//     .post(
+//       signUp,
+//       {
+//         user: {
+//           email: email,
+//           password: password,
+//           password_confirmation: passwordConfirmation,
+//         },
+//       },
+//       { withCredentials: true }
+//     )
+//     .then((response) => {
+//       if (response.data.status === "created") {
+//         props.handleSuccessfulAuthentication(response.data);
+//       }
+//     })
+//     .catch((error) => {
+//       console.log("registration error", error);
+//     });
+//   e.preventDefault();
+// };
```

Comment:
これだと、axiosでリクエストしたあと結局、レスポンスを戻り値として返せていないので
やるならこんな感じですかねー

```suggestion
// paramsにはこんな感じのオブジェクトが入る
// こういう時にTypeScriptだと型定義出来るのでとてもわかりやすい
// const params = {
// 	user: {
// 		email: email,
// 		password: password,
// 		password_confirmation: passwordConfirmation,
// 	},
// }

export const fetchSignUp = async (params) => {
	try {
		const response = await axios.post(signUp, params, {withCredentials: true})
		return response.data
	} catch (error) {
		console.error("registration error", error);
	}
};
```

Reply:
またwithcredentialsなどは、axiosのインスタンスを先に作成することで毎回書くしんどさを省略できます
https://qiita.com/terrierscript/items/ccb56b6fc05aa7821c42